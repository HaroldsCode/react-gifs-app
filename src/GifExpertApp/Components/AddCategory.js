import { useState, useEffect } from "react";
import { IoSearch } from 'react-icons/io5';
import PropTypes from "prop-types";

export const AddCategory = ({ setQuery, query }) => {

  const [inputValue, setInputValue] = useState(query);
  const [isInvalid, setIsInvalid] = useState(false);
  const [isNotEmpty, setIsNotEmpty] = useState( !!query ? true : false );

  useEffect(() => {
    setInputValue(query);
    setIsNotEmpty( !!query ? true : false );
  }, [ query ])

  const handleChange = (e) => {
    let data = e.target.value.trim();
    setInputValue(data);
    if(data.length > 0) {
      setIsNotEmpty( true );
    }else{
      setIsNotEmpty( false );
    }
  };

  const emptyInput = () => {
    setInputValue('');
    setIsInvalid(false);
    setIsNotEmpty( false );
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 0) {
      setQuery(inputValue.trim());
      localStorage.setItem("query", inputValue.trim());
      setIsInvalid(false);
      e.target.firstChild.firstChild.blur()
    } else {
      setIsInvalid(true);
    }
  };

  return (
    <form className="form" onSubmit={handleSubmit}>
      <div className="form__input_group">
        <input
          className={!isInvalid ? "form__input" : "form__input invalid"}
          type="text"
          value={inputValue}
          onChange={handleChange}
          placeholder="Escribe algo"
        />
        {isNotEmpty  && <button type="button" onClick={ emptyInput }> X </button> }
      </div>
      <button type="submit" > <IoSearch /> </button>
    </form>
  );
};

AddCategory.propTypes = {
  setQuery: PropTypes.func.isRequired,
  query: PropTypes.string.isRequired,
};
