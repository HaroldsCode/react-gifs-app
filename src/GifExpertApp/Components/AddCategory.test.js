import React from 'react';
import { shallow } from "enzyme"
import { AddCategory } from "./AddCategory"

describe('AddCategory tests', () => {
    const setNumbers = jest.fn();
    let wrapper = shallow(<AddCategory setNumbers={setNumbers}/>);

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setNumbers={setNumbers}/>);
    })

    test('should return component', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should change text box', () => {
        const input = wrapper.find('.from__input');
        input.simulate('change', { target: { value: 'Hola Harold'} });
    })
    
    test('should not post onSubmit event', () => {
        wrapper.find('.from').simulate('submit', { preventDefault(){} });
        expect(setNumbers).not.toHaveBeenCalled();
    })

    test('should call setNumbers and to clean the text box', () => {
        wrapper.find('.from__input').simulate('change', { target: { value: 'Hola Harold'} });
        wrapper.find('.from').simulate('submit', { preventDefault(){} });
        expect(setNumbers).toHaveBeenCalled();
        expect('').toBe(wrapper.find('.from__input').prop('value'));
    })
    
})
