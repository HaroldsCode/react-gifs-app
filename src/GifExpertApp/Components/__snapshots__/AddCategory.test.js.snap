// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory tests should return component 1`] = `
<form
  className="from"
  onSubmit={[Function]}
>
  <input
    className="from__input"
    onChange={[Function]}
    placeholder="Escribe algo"
    type="text"
    value=""
  />
</form>
`;
